src/components/Contacts/ContactProfileIcon.jsx

```diff
@@ -0,0 +1,64 @@
+// React Imports
+import React, { useContext } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+// Inrupt Imports
+import { getWebIdDataset } from '@inrupt/solid-client';
+// Material UI Imports
+import ContactPageIcon from '@mui/icons-material/ContactPage';
+import IconButton from '@mui/material/IconButton';
+// Custom Hook Imports
+import { useNotification } from '@hooks';
+// Context Imports
+import { DocumentListContext } from '@contexts';
+// MUI Theme
+import theme from '../../theme';
+
+/**
+ * contactProfileIconProps is an object that stores the props for the
+ * ContactProfileIcon component
+ *
+ * @typedef {object} contactProfileIconProps
+ * @property {object} contact - Contain the object that stores contact metadata
+ * @memberof typedefs
+ */
+
+/**
+ * ContactProfileIcon Component - Component that generates the contact profile
+ * icon for the Contacts List which is a special link to their profile page
+ *
+ * @memberof Contacts
+ * @name ContactProfileIcon
+ * @param {contactProfileIconProps} Props - Props for ContactProfileIcon
+ * @returns {React.JSX.Element} The ContactProfileIcon Component
+ */
+const ContactProfileIcon = ({ contact }) => {
+  const { setContact } = useContext(DocumentListContext);
+  const navigate = useNavigate();
+  const { addNotification } = useNotification();
+
+  // Event handler for profile page routing
+  const handleSelectProfile = async (contactInfo) => {
+    try {
+      await getWebIdDataset(contactInfo.webId);
+      setContact(contactInfo);
+    } catch {
+      setContact(null);
+      navigate('/contacts');
+      addNotification('error', 'WebId does not exist');
+    }
+  };
+
+  return (
+    <Link
+      to={`/profile/${encodeURIComponent(contact.id)}`}
+      state={{ contact: contact.value }}
+      style={{ textDecoration: 'none', color: theme.palette.primary.dark }}
+    >
+      <IconButton onClick={() => handleSelectProfile(contact.value)}>
+        <ContactPageIcon />
+      </IconButton>
+    </Link>
```

Comment:
This will render a `button` inside of an `a` tag. Interactive buttons shouldn't be nested because usually one of the two will be ignored. 

From an accessibility perspective, generally [`a` tags](https://www.w3.org/TR/wai-aria-1.2/#link) should be used to navigate and [buttons](https://www.w3.org/TR/wai-aria-1.2/#button) should be used for an action. In this case, I think you could add the `onClick` to the `Link` element without any adverse effects and improve the user experience.

Reply:
This is a React Router Link that helps with sending a specific state (i.e. the `contact` object) along with the redirect and browser URL. A simple anchor tag in this case would not do.

Reply:
Or do you mean to just move the onClick to Link and just leave the Icon wrapped without the IconButton? I could try that.

Reply:
Got this changed to:

```jsx
 <Link
    to={`/profile/${encodeURIComponent(contact.id)}`}
    state={{ contact: contact.value }}
    style={{ textDecoration: 'none', color: theme.palette.primary.dark }}
    onClick={() => handleSelectProfile(contact.value)}
  >
    <ContactPageIcon />
  </Link>
```

Reply:
Yeah, that's what I was thinking. Thank you!